module-4/backend/week_one.md

```diff
@@ -1,19 +1,42 @@
+
 2. What are some tools to help debug JavaScript code?
+Some tools I was previously utilizing to help debug JavaScript code include the use of `console.log()` to see what return values I get from functions and variables, `eval(pry.it)` (also, `debugger;`). After Friday's lesson I found that Chrome's developer tools (Element, Console, Source and Network tabs) are also very useful in assisting in JavaScript debugging and expanding my understanding of the performance of my applications.
+
 3. What are some tools you need in order to unit test your JavaScript?
+Some tools needed in order to unit test JavaScript are Mocha and Chai. Mocha is used for running tests and provieds `describe()`, `context()` and `it()` functionality, while Chai is a popular assertion library.
+
 4. What is the syntax for invoking a function? Give an example.
+You may have a function called `adoptCat` which does x, y and z. Simply calling `adoptCat` in the console will return that it is a function, but that function will not be invoked unless it is called like so: `adoptCat()`
+
 5. What's `this` in JavaScript?
+JavaScript's `this` refers to the parent object, and is dependent on the context in which it was referenced. When used in the global context, `this` refers to 'document' or 'window'. In the context of an object, `this` refers to and is boud to the object itself.
+
 6. What is Webpack and why is it useful?
+Webpack is a Node package that inspects our asses, finds dependencies, and creates a JS file (main.bundle.js) that is used for production.
+
 7. When/why do you want to use event delegation?
+Event delegation involves taking advantage of the fact that events bubble in the event loops by setting an event listener on one parent. This event listener analyzes bubbled events to match its child elements.
+
+One may want to use event delegation to prevent the possibility of setting mutiple listeners on the same node. Also, as was shown in class, event delegation is useful as event listeners are only applied to objects that are present within the HTML document when the script is loaded. If one applies event listeners to `li` objects but new `li` objects are added after the page has loaded, event listeners will not be connected to those new `li` objects as the script has already been run.
+
 8. What's `npm` and what do we use it for?
+`npm` is a Node Package Manager (hence n.p.m.) that allows for organization of outside packages much like Ruby Gems.
 
-#### Review  
+#### Review
 9. What's the MVC design pattern? Describe each part of MVC.
+The MVC is a design pattern that revolves around Models, Views, and Controllers.
```

Comment:
MVC is a design pattern that can be applied to other languages other than ruby/rails. I like the rails examples just be sure in an interview to say these are specific to rails